src/EcommerceStatsBundle/Resources/config/services.xml

```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" ?>
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+
+        <!-- Controller -->
+        <service
+            id="chameleon_system_ecommerce_stats.controller.csv_export"
+            class="ChameleonSystem\EcommerceStatsBundle\Controllers\CsvExportController"
+            autowire="true"
+        />
+
+        <!-- Module -->
+        <service
+            id="chameleon_system_ecommerce_stats.backend_module.ecommerce_stats"
+            class="ChameleonSystem\EcommerceStatsBundle\Bridge\Chameleon\BackendModule\EcommerceStatsBackendModule"
+            shared="false"
+            autowire="true"
+        >
+            <argument key="$translator" type="service" id="chameleon_system_core.translator" />
+            <argument key="$urlUtil" type="service" id="chameleon_system_core.util.url" />
+            <tag name="chameleon_system.module"/>
+        </service>
+
+        <service
+            id="chameleon_system_ecommerce_stats.ecommerce_stats_table"
+            class="ChameleonSystem\EcommerceStatsBundle\Service\StatsTableService"
+        >
+            <argument type="service" id="database_connection" />
+            <argument type="service" id="chameleon_system_core.translator" />
+            <argument type="service" id="logger" />
+        </service>
+        <service
+            id="chameleon_system_ecommerce_stats.csv_export"
+            class="ChameleonSystem\EcommerceStatsBundle\Service\CsvExportService"
+        >
+            <argument type="service" id="chameleon_system_core.translator" />
+        </service>
+        <service
+            id="chameleon_system_ecommerce_stats.top_seller"
+            class="ChameleonSystem\EcommerceStatsBundle\Service\TopSellerService"
+        >
+            <argument type="service" id="database_connection" />
+        </service>
+
+        <service
+            class="ChameleonSystem\EcommerceStatsBundle\Bridge\Chameleon\Twig\FormatterTwigExtension"
+            id="chameleon_system_ecommerce_stats.twig.formatter"
+        >
+            <tag name="twig.extension"/>
+        </service>
+
+        <!-- Support for autowiring interfaces -->
```

Comment:
What is autowiring (here)?

Reply:
You can have symfony autowire your services for you - which basically means instead of having to specify each dependency manually in `services.xml` it looks at the type annotations of your constructor and figures it out.

Check `CsvExportController` in this PR for example: The service definition contains no arguments but `autowire="true"`. Symfony looks at the class and sees the constructor

```php
    public function __construct(
        CsvExportServiceInterface $csvExportService,
        StatsTableServiceInterface $statsTableService,
        TopSellerServiceInterface $topSellerService
    )
``` 

and figures out what to inject. The whole autowiring setup however depends on services (or service aliases) to exist for the FQN of the interfaces, which is what is added here.

If you setup a new symfony project today, autowiring has become the defacto standard and can, if all dependency support i, save a lot of time when developing.